### ISSUE TITLE
Set locale variables automatically for the integrated terminal

### ISSUE THREAD

**Issue by ibratoev** [Microsoft/vscode#7301]
- VSCode Version: 1.2.0
- OS Version: OSX 10.11.5

Steps to Reproduce:
1. I use ZSH for shell.
2. Open Integrated Terminal and run `locale`.
3. Locale is not set properly. Here is my result:

```
LANG=
LC_COLLATE="C"
LC_CTYPE="C"
LC_MESSAGES="C"
LC_MONETARY="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_ALL=
```

Expected:

An optional similar to the default Terminal app in OSX. It makes sense to have it ON by default.
Option from Terminal settings:
<img width="648" alt="screen shot 2016-06-07 at 10 11 48 am" src="https://cloud.githubusercontent.com/assets/62861/15848929/5d36298c-2c98-11e6-801f-9e0a52937a74.png">

Workaround:

Add the following to your profile (`~/.zhrc` for ZSH):

```
export LC_ALL=en_US.UTF-8
export LANG=en_US.UTF-8
```

**Comment by Tyriar** [comment]
This might be OS X-specific, seems fine in Linux (vscode/bash left, gnome-terminal/bash right)

![image](https://cloud.githubusercontent.com/assets/2193314/15849679/96c40b10-2c48-11e6-80f7-f8753e26f2e3.png)

**Comment by Tyriar** [comment]
If anyone has insight on how locale works on Mac it would be appreciated, PRs welcome.

**Comment by satokaz** [comment]
VSCode Version: 1.4.0 Insiders (commit 217e0b5
OS Version: OS X 10.11.5（15F34）

Shell: /bin/bash, 
LANG variable is not explicitly set. It is automatically set by the system.
following is a result of running the `locale` command on the OS X Terminal (not vscode terminal).
```bash
Kazuyuki-no-MacBook-Pro:~ satokaz$ locale
LANG="ja_JP.UTF-8"
LC_COLLATE="ja_JP.UTF-8"
LC_CTYPE="ja_JP.UTF-8"
LC_MESSAGES="ja_JP.UTF-8"
LC_MONETARY="ja_JP.UTF-8"
LC_NUMERIC="ja_JP.UTF-8"
LC_TIME="ja_JP.UTF-8"
LC_ALL=
```

Please tell me the behavior of expected "terminal.integrated.setLocaleVariables" setting.

In the Japanese locale, it will be set as follows.
 `"terminal.integrated.setLocaleVariables":true` (default) :

```bash
bash-3.2$ locale
LANG="ja.UTF-8"
LC_COLLATE="C"
LC_CTYPE="C"
LC_MESSAGES="C"
LC_MONETARY="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_ALL=
```

"Ja.UTF-8" is not correct.
"Ja_JP.UTF-8" is correct.



## "terminal.integrated.setLocaleVariables": true

<img width="1606" alt="2016-07-13 6 18 28" src="https://cloud.githubusercontent.com/assets/13635244/16784130/48630b9c-48c2-11e6-842b-da4388934809.png">

## "terminal.integrated.setLocaleVariables": false

This is the value you expected Looks.


![2016-07-13 6 19 44](https://cloud.githubusercontent.com/assets/13635244/16784135/4ec54fae-48c2-11e6-9364-75649ff8a12d.png)

It looks like when the `"terminal.integrated.setLocaleVariables":false` is set to the correct value has been set.

**Comment by Tyriar** [comment]
Ping @chrmarti, sometimes vscode's locale is only the language, not dialect eg. "ja". This is likely why it's set to ja, not ja_JP. As for why it's working on this version of OS X, I have no idea?

**Comment by chrmarti** [comment]
We use electron's app.getLocale() to read the locale, which limits it's values to those electron has translations for, that includes 'ja', but not 'ja_JP': http://electron.atom.io/docs/api/app-locales/

We could use some other means to get at the full locale identifier (like https://www.npmjs.com/package/os-locale).

**Comment by chrmarti** [comment]
@dbaeumer, would it make sense to make platform.locale be the full OS's locale rather than just what electron supports? Having that will make sense for other (than shells) external processes that support more locales than electron too.

**Comment by satokaz** [comment]
Thank you for the details.

It is considered to be difficult to derive the OS loclae from Electron's app.getLocale().
short locale is also not supported.
In addition, can not care of the LC_* environment variables.

It is desirable to be automatic in the locale setting, but this implementation is thought that only lead to confusion.

Currently, That prompt the user to set the `export LANG=""` environment variable to the configuration file of the shell (.zprofile, .zshrc, .bash_profile, .bashrc ...) to use, I think it is good to describe to the FAQ.

Microsoft/vscode project is great.

Thanks,
Kazuyuki

**Comment by Tyriar** [comment]
This should be fixed in 1.10

**Comment by dbaeumer** [comment]
@chrmarti I decided against this since the npm modules I looked at when implementing this all spawn processes and this is on the critical start up path and we can't load any code without knowing which locale to use. IMO if someone really needs the real OS locale then they can use a corresponding npm module.
